txtl/sbmlutil.py

```diff
@@ -0,0 +1,230 @@
+        species.setConstant(False)
+        species.setBoundaryCondition(False)
+        species.setHasOnlySubstanceUnits(False) #! TODO: confirm this is OK
+
+    else:
+        if debug: print("add_species: %s already exists", species.getId())
+        
+    # Set the initial concentration (if specified)
+    #! TODO: Decide whether to warn if species is already present
+    #! TODO: add initial concentrations if reaction is present
+    if ic != None:
+        if debug: print("    %s IC = %s" % (species_name, ic))
+        species.setInitialConcentration(float(ic))
+
+    return species
+
+# Look for a species in the current mixture
+def find_species(mixture, species_name):
+    model = mixture.model   # Get the model where we will store results
+    
+    # Construct the species ID (no-op if already a species ID)
+    species_id = re.sub(" ", "_", species_name)
+    species_id = re.sub("--", "_", species_id)
+    species_id = re.sub(":", "_", species_id)
+    
+    return model.getSpecies(species_id)
+
+# Helper function to add a pameter to the model
+def add_parameter(mixture, name, value=0, debug=False):
+    model = mixture.model   # Get the model where we will store results
+
+    # Check to see if this parameter is already present
+    parameter = find_parameter(mixture, name)
+    if parameter == None:
+        if debug: print("Adding parameter %s" % name)
+        parameter = model.createParameter()
+        parameter.setId(name)
+
+    else:
+        if debug: print("add_parameter: %s already exists", parameter.getId())
+
+    # Set the value of the parameter
+    parameter.setValue(float(value))
+    parameter.setConstant(True)
+
+    return parameter
+
+# Look for a parameter in the current model
+def find_parameter(mixture, id):
+    model = mixture.model   # Get the model where we will store results
+    
+    return model.getParameter(id)
+
+# Helper function to add a reaction to a model
+def add_reaction(mixture, reactants, products, kf, kr=None, id=None,
+                 parameters={}, prefix="r", debug=False):
+    """Add a reaction to a model
+
+    The `add_reaction` function is used to add a reaction to a model.
+    It allows specifation for reaction rates as either symbolic or
+    numeric entries, including the use of global symbolic names.
+    Reactions can be unidirectional or reversible.
+
+    Parameters
+    ----------
+    model       SBML model
+    reactants   List of SBML species that are reactants in the reaction
+    projects    List of SBML species that are products of the reaction
+    kf          Forward rate constant (parameter, string, number, or list)
+    kf          Reverse rate constant (None if non-reversible)
+    id          Optional parameter to specify reaction id (otherwise numbered)
+
+    Note: the current implementation requires that non-unitary
+    stochiometries be represented by repeated entries in the reactants
+    and/or products list.
+
```

Comment:
Maybe we can have an optional argument after parameters argument for a stoichiometry vector to address this ?

Reply:
Agree.  I've added a comment to this effect, but don't think it is needed for v0.1.